### PR TITLE
[Merged by Bors] - chore: code cleanup

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation

I removed some dead code from the codebase and cleaned up some other issues my linter complained about.

## Description

- Prepared changelog for changes of the next release
- un-exported errors that don't need to be exported
- removed assignments to `_` if all return values are ignored
- removed dead code from the `common/util` package

## Test Plan

existing tests pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: common/util/json.go [hunk 1]
```diff
@@ -23,57 +23,18 @@ import (
...
 	"fmt"
 	"reflect"
-	"strconv"
 )
 
-var bytesT = reflect.TypeOf(Bytes(nil))
-
 // Bytes marshals/unmarshals as a JSON string with 0x prefix.
 // The empty slice marshals as "0x".
 type Bytes []byte
```
Comment (ivan4th) [inline common/util/json.go]: Hmm it looks like `Bytes` is not actually used anywhere except `common/types/hashes.go`, where its usage is also removed in this patch. So probably we may remove it altogether

**Review by ivan4th — APPROVED** [review]
Good except that we probably need to remove that `Bytes` type altogether

File: common/util/json.go (hunk 1 above)
Comment (fasmat) [reply common/util/json.go]: Removed `Bytes` and the `Encode` function as well. What's left in `common/util` is still used, some of which in critical places and should probably be looked at in a different PR to ensure it's properly handled 🙂

**Comment by fasmat** [comment]
bors merge

**Review by poszu — APPROVED** [review]

**Pull request closed without merging**